server/db/models/order.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+var mongoose = require('mongoose');
+var Promise = require('bluebird');
+
+var ObjectId = mongoose.Schema.Types.ObjectId;
+
+var deepPopulate = require('mongoose-deep-populate')(mongoose);
+
+var schema = new mongoose.Schema({
+    status: {
+        type: String,
+        enum: ['shipped', 'confirmed', 'pending', 'cancelled'],
+        required: true,
+        default: 'pending'
+    },
+    items: {
+        type: [{
+            quantity: {
+                type: Number,
+                required: true
+            },
+            product: {
+                type: ObjectId,
+                ref: 'Product',
+                required: true
+            }
+        }],
+        required: true
+    },
+    date: {
+        type: Date,
+        default: Date.now(),
+        required: true
+    },
+    user: {
+        type: ObjectId,
+        ref: 'User'
+    },
+    session: {
+        type: String
+    }
+
+});
+
+schema.plugin(deepPopulate);
+
+// schema.path('user').validate(function (value) {
+//     return !this.session;
+// }, 'session stuff');
+//
+// schema.path('session').validate(function (value) {
+//     return !this.user;
+// });
+
```

Comment:
What if it has a session and a user property? I think by the spec it should have either one or the other.. that's why I put these in there.